tests/test_umbral.py

```diff
@@ -24,28 +26,40 @@
 ]
 
 
-def test_decapsulation_by_alice():
-    priv_key = umbral.gen_priv()
-    pub_key = umbral.priv2pub(priv_key)
+@pytest.fixture(scope='function')
+def alices_keys(curve=default_curve()):
+    params = UmbralParameters(curve=curve)
+    priv = keys.UmbralPrivateKey.gen_key(params)
+    pub = priv.get_pub_key()
+    TestKeyPair = namedtuple('TestKeyPair', 'name priv pub')
+    return TestKeyPair('alice', priv, pub)
+
+
+@pytest.fixture(scope='function')
+def bobs_keys(curve=default_curve()):
+    params = UmbralParameters(curve=curve)
+    priv = keys.UmbralPrivateKey.gen_key(params)
+    pub = priv.get_pub_key()
+    TestKeyPair = namedtuple('TestKeyPair', 'name priv pub')
+    return TestKeyPair('bob', priv, pub)
+
 
-    sym_key, capsule = umbral._encapsulate(pub_key)
+def test_decapsulation_by_alice(alices_keys):
+    name, alice_priv, alice_pub = alices_keys
+
+    sym_key, capsule = umbral._encapsulate(alice_pub.point_key)
     assert len(sym_key) == 32
 
     # The symmetric key sym_key is perhaps used for block cipher here in a real-world scenario.
-    sym_key_2 = umbral._decapsulate_original(priv_key, capsule)
+    sym_key_2 = umbral._decapsulate_original(alice_priv.bn_key, capsule)
     assert sym_key_2 == sym_key
 
 
 @pytest.mark.parametrize("N, M", parameters)
-def test_simple_api(N, M, curve=default_curve()):
-
+def test_simple_api(alices_keys, bobs_keys, N, M, curve=default_curve()):
     params = UmbralParameters(curve=curve)
-
-    priv_key_alice = keys.UmbralPrivateKey.gen_key(params)
-    pub_key_alice = priv_key_alice.get_pub_key()
-
-    priv_key_bob = keys.UmbralPrivateKey.gen_key(params)
-    pub_key_bob = priv_key_bob.get_pub_key()
+    _, priv_key_alice, pub_key_alice = alices_keys
```

Comment:
I know it's from a fixture, but should we have `_something_something` instead of `_`?

Reply:
OK, good idea.  In this case it's unpacking the keypair's name from the namedtuple
`TestKeyPair = namedtuple('TestKeyPair', 'name priv pub')`.

Reply:
I ended up removing the TestKeyPair name attr. all together.